### PR TITLE
[Side Channels] Edited the documenation and renamed a few things

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

Update to the documentation for #3807

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [x] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [x] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [x] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [x] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: com.unity.ml-agents/CHANGELOG.md [hunk 1]
```diff
@@ -37,6 +37,11 @@ and this project adheres to
 - The obsolete `Agent` methods `GiveModel`, `Done`, `InitializeAgent`,
   `AgentAction` and `AgentReset` have been removed.
 - The GhostTrainer has been extended to support asymmetric games and the asymmetric example environment Strikers Vs. Goalie has been added.
+- The SideChannel API has changed:
+  - `EnvironmentParameters` replaces the default `FloatPropertiesChannel`. You can access the `EnvironmentParameters` with `Academy.Instance.EnvironmentParameters` on C# and create an `EnvironmentParametersChannel` on Python
+  - `SideChannelUtils` was renamed `SideChannelManager`
+  - The `Academy` instance now has a `StatsRecorder` property
```
Comment (mmattar) [inline com.unity.ml-agents/CHANGELOG.md:43]: Do we want to emphasize that the StatsSideChannel is now internal and that this is the new way

File: docs/Migrating.md [hunk 2]
```diff
@@ -33,6 +33,11 @@ double-check that the versions are in the same. The versions can be found in
 - The signature of `Agent.Heuristic()` was changed to take a `float[]` as a
   parameter, instead of returning the array. This was done to prevent a common
   source of error where users would return arrays of the wrong size.
+- The SideChannel API has changed:
```
Comment (mmattar) [inline docs/Migrating.md:36]: We need to mention the user-facing changes of going from StatsSideChannel to Academy.Instance.StatsRecorder

Comment (mmattar) [reply docs/Migrating.md:36]: And that StatsSideChannel is now internal

Comment (mmattar) [inline docs/Migrating.md:36]: Should we add that EngineConfigurationChannel is no internal

File: com.unity.ml-agents/CHANGELOG.md (hunk 1 above)
Comment (mmattar) [reply com.unity.ml-agents/CHANGELOG.md:43]: Also mention that EngineConfigurationChannel is now internal?

Comment (mmattar) [reply com.unity.ml-agents/CHANGELOG.md:43]: Worth mentioning that `OnMessageReceived` is now internal?

File: docs/Migrating.md (hunk 2 above)
Comment (mmattar) [reply docs/Migrating.md:36]: Worth mentioning that `OnMessageReceived` is now internal?

File: docs/Migrating.md [hunk 3]
```diff
@@ -33,6 +33,11 @@ double-check that the versions are in the same. The versions can be found in
...
+  - `EnvironmentParameters` replaces the default `FloatPropertiesChannel`. You can access the `EnvironmentParameters` with `Academy.Instance.EnvironmentParameters`
+  - `SideChannelUtils` was renamed `SideChannelManager`
+  - The `Academy` instance now has a `StatsRecorder` property
+  - `SideChannelManager.GetSideChannel(s)` has been removed from the API
```
Comment (mmattar) [inline docs/Migrating.md:40]: Also UnregisterAllSideChannels

File: com.unity.ml-agents/CHANGELOG.md (hunk 1 above)
Comment (chriselion) [reply com.unity.ml-agents/CHANGELOG.md:43]: StatsSideChannel hasn't actually been released yet, so we should just edit the existing line for it.

File: com.unity.ml-agents/CHANGELOG.md [hunk 4]
```diff
@@ -37,6 +37,11 @@ and this project adheres to
...
+  - `SideChannelManager.GetSideChannel(s)` has been removed from the API
```
Comment (chriselion) [inline com.unity.ml-agents/CHANGELOG.md:44]: Same - SideChannelManager/Utils wasn't in 0.15, so we don't need to mention it here.

File: docs/Migrating.md [hunk 5]
```diff
@@ -54,6 +59,10 @@ double-check that the versions are in the same. The versions can be found in
 - If your Agent class overrides `Heuristic()`, change the signature to
   `public override void Heuristic(float[] actionsOut)` and assign values to
   `actionsOut` instead of returning an array.
+- If you used `SideChannels` you must:
```
Comment (chriselion) [inline docs/Migrating.md:62]: Revise the "Replace `Academy.FloatProperties`..." lines above instead.

File: docs/Python-API.md [hunk 6]
```diff
@@ -306,41 +306,34 @@ i = env.reset()
...
 ...
 ```
 
 Once a property has been modified in Python, you can access it in C# after the next call to `step` as follows:
 
 ```csharp
-var sharedProperties = SideChannelUtils.GetSideChannel<FloatPropertiesChannel>();
-float property1 = sharedProperties.GetPropertyWithDefault("parameter_1", 0.0f);
+var envParameters = Academy.Instance.EnvironmentParameters;
+float property1 = envParameters.GetPropertyWithDefault("parameter_1", 0.0f);
```
Comment (chriselion) [inline docs/Python-API.md:336]:
```suggestion
float property1 = envParameters.GetWithDefault("parameter_1", 0.0f);
```

File: com.unity.ml-agents/CHANGELOG.md [hunk 7]
```diff
@@ -36,21 +32,30 @@ and this project adheres to
...
+  asymmetric example environment Strikers Vs. Goalie has been added.
+- The SideChannel API has changed (#3833, #3660) :
+  - Introduced the `SideChannelManager` to register, unregister and access side
+  channels.
+  - `EnvironmentParameters` replaces the default `FloatProperties`.
+  You can access the `EnvironmentParameters` with
+  `Academy.Instance.EnvironmentParameters` on C# and create an
+  `EnvironmentParametersChannel` on Python
+  - `SideChannel.OnMessageReceived` is now a protected method (was public)
+  - The `Academy` instance now has a `StatsRecorder` property
```
Comment (chriselion) [inline com.unity.ml-agents/CHANGELOG.md:45]:
```suggestion
```
Redundant because of the "Added a feature to allow..." line below.

**Pull request merged**